[BE4] 노동희

nodonghui opened the pull request:
리뷰 잘 부탁드립니다!

nodonghui committed:
add wiseSaying

nodonghui committed:
Merge remote-tracking branch 'upstream/main'

vdvhk12 commented on src/main/java/com/llwiseSaying/Controller/WiseSayingController.java:
```diff
@@ -0,0 +1,195 @@
+        //최대 페이지수를 넘지않고 설정
+        page=Math.min(totalPage,page);
+
+        wiseSayingList.stream()
+                .forEach(s -> System.out.println(s));
```
Service에서 List<String>으로 바로 출력 가능한 형태로 가져오고, Controller에서는 바로 출력만 해주는게 깔끔한 것 같네요.
저는 List<WiseSaying> 타입으로 리턴 받아서 Controller에서 getId ~ 로 출력하는데.. 보면서 띠용 :astonished: 하고 갑니다!

vdvhk12 commented on src/main/java/com/llwiseSaying/WiseSaying.java:
```diff
@@ -0,0 +1,38 @@
+        return writer;
+    }
+
+    public String toString() {
+        String printWiseSaying=id + " / " + writer + " / " + content;
+        return printWiseSaying;
+    }
```
확실히 메서드로 만들어서 관리하는게 깔끔하고 보기 좋네요
저도 참고해서 수정 해야겠습니다..ㅎㅎ

vdvhk12 commented on src/test/java/WiseSayingControllerTest.java:
```diff
@@ -0,0 +1,288 @@
+
+    ByteArrayOutputStream output;
+
+    @BeforeEach
+    void beforeInit(){
+        output = AppUtil.setOutToByteArray();
+        DBdirectoryPath="db/test/wiseSaying";
+    }
+
+    @AfterEach()
+    void afterInit() {
+        AppUtil.clearSetOutToByteArray(output);
+        DBdirectoryPath="db/wiseSaying";
+    }
```
테스트할 때 파일 저장 경로를 시작 전에 test저장 경로로 바꾸고 다시 메인 파일 저장 경로로 바꾸시는군요 :open_mouth:
저는 이렇게 할 생각이 안 떠올라서 아직 수동으로 바꾸고 있는데 하나 배우고 갑니다 :thumbsup:

vdvhk12 commented:
전체적으로 코드가 간결하고 가독성이 좋은 것 같습니다.  
그래서 위에서 부터 읽어 내려가는데 막히는 부분 없이 술술 읽혔던 것 같습니다. 
코드를 보면서 제 코드를 리팩토링 할때 참고 할만한 내용도 있었고, 많이 배우고 갑니다. 잘봤습니다. :blush:
주말 잘 보내세요~

nodonghui committed:
refactor: config 파일 추가와 init과정 개선

리포지터리에 저장해 사용하던 데이터베이스 경로와 저장 파일 정보를 config파일로
분리했습니다.
순서가 난잡하던 init과정을 개선했습니다.
